src/azure/accountService.ts

```diff
@@ -65,40 +78,23 @@ export class AccountService {
     }
 
     public async createSecurityTokenMapping(): Promise<any> {
-        if (!this._token) {
-            let promise = new Deferred();
-            this._token = this._session.credentials.getToken((error, result ) => {
-                if (result) {
-                    this._isStale = false;
-                    this._token = result;
-                }
-                if (error) {
-                    this._isStale = true;
-                }
-                promise.resolve();
-            });
-            await promise;
-        }
         let mapping = {};
-        mapping[this._session.tenantId] = {
-            expiresOn: this._token.expiresOn.toISOString(),
-            resource: this._token.resource,
-            tokenType: this._token.tokenType,
-            token: this._token.accessToken
+        mapping[this.getHomeTenant(this.account).id] = {
```

Comment:
Please use a `Map<string, Token>` here and make the return type the same

Reply:
it will be a Map<string, string> since refreshToken returns a string